### PR TITLE
Add Plonky2 verifier

### PR DESCRIPTION
**Pull request by esciiee**

### CODE REVIEW

File: accept_handlers/handle_plonky2.go [hunk 1]
```diff
@@ -27,17 +28,20 @@ func HandlePlonky2(
 	baseDir string,
 	endPointRequester *requester.EndpointRequester,
 ) error { //@todo send the hashes stored for every proofvaltype to rust server
-	elfKey := storage.DeployKey(imageID, elfValType)
+	commonDataKey := storage.DeployKey(imageID, commonDataValType)
+	verifierDataValType := storage.DeployKey(imageID, verifierDataValType)
```
Comment (esciiee) [inline accept_handlers/handle_plonky2.go:32]: change verifierDataValType to verifierDataValKey

File: accept_handlers/handle_plonky2.go [hunk 2]
```diff
@@ -27,17 +28,20 @@ func HandlePlonky2(
...
 	proofKey := storage.DeployKey(imageID, proofValType)
-	elfFilePath := baseDir + "/" + elfKey
+	commonDataFilePath := baseDir + "/" + commonDataKey
+	verifierDataFilePath := baseDir + "/" + verifierDataValType
```
Comment (esciiee) [inline accept_handlers/handle_plonky2.go:35]: Here too change verifierDataValType to verifierDataValKey

File: actions/plonky2.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	"github.com/sausaging/hypersdk/consts"
+	"github.com/sausaging/hypersdk/state"
+)
+
+var _ chain.Action = (*PLONKY2)(nil)
+
+type PLONKY2 struct {
+	ImageID       ids.ID `json:"image_id"`
+	ProofValType  uint64 `json:"proof_val_type"`
+	TimeOutBlocks uint64 `json:"time_out_blocks"`
```
Comment (manojkgorle) [inline actions/plonky2.go:22]: Add commonDataValType & verifierValType here

File: accept_handlers/consts.go [hunk 4]
```diff
@@ -5,6 +5,9 @@ const (
 	MIDENVERIFY    = 2
 	RISCZEROVERFIY = 3
 	JOLTVERIFY     = 4
+	PLONKY2VERIFY  = 5
 )
 
 const elfValType = 1
+const commonDataValType = 2
+const verifierDataValType = 3
```
Comment (manojkgorle) [inline accept_handlers/consts.go:13]: Don't hard code commonDataValType & verifierDataValType, let user decide the values and pass with the plonky2 action

File: accept_handlers/handle_plonky2.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+type Plonky2ReplyArgs struct {
+	IsSubmitted bool `json:"is_submitted"`
+}
+
+func HandlePlonky2(
+	txID ids.ID,
+	imageID ids.ID,
+	proofValType uint16,
```
Comment (manojkgorle) [inline accept_handlers/handle_plonky2.go:27]: add commonDataValType and verifierDataValType here too

File: cmd/morpheus-cli/cmd/testing.go [hunk 6]
```diff
@@ -183,17 +183,23 @@ var verifyCmd = &cobra.Command{
...
-		} else if verifyType == 4 {
-
-		} else if verifyType == 5 {
+		}  else if verifyType == 4 {
 			action = &actions.Jolt{
 				ImageID:       imageID,
 				ProofValType:  uint64(valType),
 				TimeOutBlocks: uint64(timeOutBlocks),
 			}
+		} else if verifyType == 5 {
```
Comment (manojkgorle) [inline cmd/morpheus-cli/cmd/testing.go:192]: ask commonDataValType and verifierDataValType

File: controller/controller.go [hunk 7]
```diff
@@ -192,6 +192,12 @@ func (c *Controller) Accepted(ctx context.Context, blk *chain.StatelessBlock) er
 			if err := handle.HandleJolt(tx.ID(), jolt.ImageID, uint16(jolt.ProofValType), c.fileDB.BaseDir(), c.config.Client); err != nil {
 				c.inner.Logger().Info("error handling Jolt", zap.Error(err))
 			}
+		case *actions.PLONKY2:
+			plonky2 := tx.Action.(*actions.PLONKY2)
+			c.trustless.ListenActions(tx.ID(), plonky2.TimeOutBlocks)
+			if err := handle.HandlePlonky2(tx.ID(), plonky2.ImageID, uint16(plonky2.ProofValType), c.fileDB.BaseDir(), c.config.Client); err != nil {
```
Comment (manojkgorle) [inline controller/controller.go:198]: same suggestion as above

File: actions/plonky2.go [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+func (*PLONKY2) MaxComputeUnits(chain.Rules) uint64 {
+	return PLONKY2ComputeUnits
+}
+
+func (s PLONKY2) Size() int {
+	return consts.IDLen + consts.Uint64Len*2
+}
+
+func (s *PLONKY2) Marshal(p *codec.Packer) {
```
Comment (manojkgorle) [inline actions/plonky2.go:51]: marshal commondatavaltype and verifierdatavaltype

File: actions/plonky2.go [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+	p.PackID(s.ImageID)
+	p.PackUint64(s.ProofValType)
+	p.PackUint64(s.TimeOutBlocks)
+}
+
+func UnmarshalPLONKY2(p *codec.Packer, _ *warp.Message) (chain.Action, error) {
+	var PLONKY2 PLONKY2
+	p.UnpackID(true, &PLONKY2.ImageID)
+	PLONKY2.ProofValType = p.UnpackUint64(true)
+	PLONKY2.TimeOutBlocks = p.UnpackUint64(true)
```
Comment (manojkgorle) [inline actions/plonky2.go:61]: unmarshall commondatavaltype and verifierdatavaltype

**Review by manojkgorle — APPROVED** [review]
thanks

**Pull request merged**